pkg/apis/extensions/v1beta1/types.go

```diff
@@ -36,6 +36,14 @@ type ScaleStatus struct {
 
 	// label query over pods that should match the replicas count. More info: http://releases.k8s.io/HEAD/docs/user-guide/labels.md#label-selectors
 	Selector map[string]string `json:"selector,omitempty"`
+
+	// label selector for pods that should match the replicas count. This is a serializated
+	// version of both map-based and more expressive set-based selectors. This is done to
+	// avoid introspection in the clients. The string will be in the same format as the
+	// query-param syntax. If the target type only supports map-based selectors, both this
+	// field and map-based selector field are populated.
+	// More info: http://releases.k8s.io/HEAD/docs/user-guide/labels.md#label-selectors
+	TargetSelector string `json:"targetSelector,omitempty"`
```

Comment:
@liggitt The structured form is good for users writing it by hand, but is painful for clients that then need to figure out how to serialize it for list/watch, especially as we evolve the API. I want to move to mirroring selectors as serialized strings in status for all APIs with selectors. Clients should be able to just send it back to apiserver uninterpreted.

Discussed in the following comment and elsewhere:
https://github.com/kubernetes/kubernetes/issues/3676#issuecomment-147524114